core/include/vecmem/containers/data/jagged_vector_buffer.hpp

```diff
@@ -93,8 +94,9 @@ class jagged_vector_buffer : public jagged_vector_view<TYPE> {
     ///        be host accessible.
     /// @param host_access_resource An optional host accessible memory
     ///        resource. Needed if @c resource is not host accessible.
-    jagged_vector_buffer(const std::vector<std::size_t>& sizes,
-                         const std::vector<std::size_t>& capacities,
+    template <typename size_t = std::size_t>
```

Comment:
I would make this slightly more elaborate...

First off, I don't think we need a default type. The compiler should be able to deduce the type from the arguments that it receives for the constructor.

But second, it could be nice to use SFINAE to only allow integer types to be used. :wink:

Reply:
And only unsigned integers, please!

Reply:
I'll add the checks for uint types only.
I put the default type in to make all the CI tests work as-is. e.g. 
https://github.com/acts-project/vecmem/blob/main/tests/cuda/test_cuda_jagged_vector_view.cpp#L170-L171
And I don't really see how it can harm us

Reply:
Okay, that's a good point. Let's keep the `std::size_t` type by default then. (I'm a bit surprised that C\+\+ didn't default to an `std::vector<int>`  type in that case though. Those number literals should've defaulted to `int`... :thinking:)